src/org/intellij/erlang/compilation/ErlangPrepareDependenciesCompileTask.java

```diff
@@ -127,162 +128,147 @@ private static ErlangModuleBuildOrderDescriptor getModuleBuildOrderInner(Module
     ErlangModuleBuildOrderDescriptor buildOrder = new ErlangModuleBuildOrderDescriptor();
     ErlangFacet erlangFacet = ErlangFacet.getFacet(module);
     List<String> globalParseTransforms = erlangFacet != null ? erlangFacet.getConfiguration().getParseTransforms() : ContainerUtil.<String>emptyList();
+    PsiManager psiManager = PsiManager.getInstance(module.getProject());
     buildOrder.myModuleName = module.getName();
-    buildOrder.myOrderedErlangModulePaths = getTopologicallySortedErlangModulePaths(ErlangModulesUtil.getErlangModules(module, false), globalParseTransforms);
-    buildOrder.myOrderedErlangTestModulePaths = getTopologicallySortedErlangModulePaths(ErlangModulesUtil.getErlangModules(module, true), ContainerUtil.<String>emptyList());
+    buildOrder.myOrderedErlangFilePaths = getTopologicallySortedErlangFilesPaths(ErlangModulesUtil.getVirtualFilesForErlangModules(module, false),
+                                                                                 ErlangModulesUtil.getVirtualFilesForErlangHeaders(module, false),
+                                                                                 globalParseTransforms,
+                                                                                 psiManager);
+    buildOrder.myOrderedErlangTestFilePaths = getTopologicallySortedErlangFilesPaths(ErlangModulesUtil.getVirtualFilesForErlangModules(module, true),
+                                                                                     ErlangModulesUtil.getVirtualFilesForErlangHeaders(module, true),
+                                                                                     globalParseTransforms,
+                                                                                     psiManager);
```

Comment:
Doesn't it feel like making these calls to ErlangModulesUtil.get*** inside getTopologicallySortedErlangFilesPaths would make the code easier to read? It would also render obtaining psiManager in this method unnecessary.